syntax = "proto3";

package ProtoTransport;

message Point {
    double x = 1;
    double y = 2;
};

message Rgb {
    uint32 red =1;
    uint32 green = 2;
    uint32 blue = 3;
};

message Rgba {
    uint32 red =1;
    uint32 green = 2;
    uint32 blue = 3;
    double alpha = 4;
};

message Color {
    oneof color_variant {
        string as_string = 1;
        Rgb as_rgb = 2;
        Rgba as_rgba = 3;
    }
}

message RenderSettings {
    double width =1;
    double height = 2;
    double padding = 3;
    double stop_radius = 4;
    double line_width = 5;
    int32 stop_label_font_size = 6;
    Point stop_label_offset = 7;
    Color underlayer_color = 8;
    double underlayer_width = 9;
    int32 bus_label_font_size = 10;
    Point bus_label_offset = 11;
    repeated Color color_palette = 12;
    repeated string layers = 13;
    double outer_margin = 14; 
    double company_radius = 15; 
    double company_line_width = 16; 
};

message Compressed {
    string name = 1;
    double lat = 2;
    double lon = 3;
}

message Renderer {
    repeated Compressed stops = 1;
    repeated Compressed companies = 2;
    RenderSettings settings = 3;
}